subprojects/core/src/main/java/org/gradle/initialization/layout/BuildTreeLocations.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.initialization.layout;
+
+import org.gradle.internal.service.scopes.Scope;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+import java.io.File;
+
+/**
+ * Locations required to set up and manage a build tree.
+ * <p>
+ * The locations are fixed even for a build-session scope, because they can't change during a continuous build.
+ */
+@ServiceScope(Scope.BuildSession.class)
```

Comment:
💭 Seems odd the scope isn't BuildTree.

Reply:
The build-session scope is a given, because that's where the information becomes available as part of the invocation lifecycle. This is also explained in the javadoc.

It's mostly a question of naming. An alternative would be to call this `BuildSessionLocations`, but this has much worse discoverability IMO as the build-tree. Plus, the data this holds is about the build tree.